appcelerator/titanium_mobile
# [TIMOB-25442] Android: Updated Titanium SDK and build tools to compile with Java 7

**JIRA:**
- https://jira.appcelerator.org/browse/TIMOB-25442
- https://jira.appcelerator.org/browse/TIMOB-25448

**Summary:**
- Updated core Android Titanium SDK code to use Java 7. (Was limited to Java 6.)
- Updated CLI to build Android modules with Java 7. (Was limited to Java 6.)

---

**Titanium SDK Java 7 Test:**

1. Build/package/install the "titanium_mobile" SDK for Android.
2. Verify that build succeeded without any Java compiler errors. _(Note that "TiApplication.java" is now using a Java 7 try-resource feature which would cause a compiler error in Java 6.)_
3. Build and run a Titanium app on Android.
4. Observe the Android log on startup and verify that a message similar to the below gets printed on startup.  _(The date will match the time you compiled the SDK.)_
```[INFO] :   TiApplication: (main) [43,43] Titanium 7.0.0 (2017/10/26 15:22 undefined)```

---

**Module Java 7 Test:**

1. Build/package/install the "titanium_mobile" SDK for Android.
2. Run the test described in [TIMOB-25448](https://jira.appcelerator.org/browse/TIMOB-25448) on Mac.
3. Run the same test on Windows.
